core/state_connector_test.go

```diff
@@ -0,0 +1,268 @@
+	to         common.Address
+	gas        uint64
+	gasPrice   *big.Int
+	gasFeeCap  *big.Int
+	gasTipCap  *big.Int
+	value      *big.Int
+	data       []byte
+	accessList types.AccessList
+}
+
+func (m *mockMessage) From() common.Address         { return m.from }
+func (m *mockMessage) To() *common.Address          { return &m.to }
+func (m *mockMessage) GasPrice() *big.Int           { return m.gasPrice }
+func (m *mockMessage) GasFeeCap() *big.Int          { return m.gasFeeCap }
+func (m *mockMessage) GasTipCap() *big.Int          { return m.gasTipCap }
+func (m *mockMessage) Value() *big.Int              { return m.value }
+func (m *mockMessage) Gas() uint64                  { return m.gas }
+func (m *mockMessage) Nonce() uint64                { return 0 }
+func (m *mockMessage) Data() []byte                 { return m.data }
+func (m *mockMessage) AccessList() types.AccessList { return m.accessList }
+func (m *mockMessage) IsFake() bool                 { return true }
+
+type mockStateConnectorCaller struct {
+	context  vm.BlockContext
+	msg      *mockMessage
+	CallFunc func(context vm.BlockContext, caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error)
+}
+
+func (m *mockStateConnectorCaller) Call(caller vm.ContractRef, addr common.Address, input []byte, gas uint64,
+	value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+
+	return m.CallFunc(m.context, caller, addr, input, gas, value)
+}
+
+func (m *mockStateConnectorCaller) WithBlockContext(bc vm.BlockContext) { m.context = bc }
+
+func (m *mockStateConnectorCaller) BlockContext() vm.BlockContext { return m.context }
+
+func buildStateConnectorMock(from string, addr common.Address) (*mockMessage, *mockStateConnectorCaller) {
+	msg := mockMessage{
+		from: common.BytesToAddress([]byte(from)),
+	}
+	scc := mockStateConnectorCaller{
+		msg: &msg,
+		context: vm.BlockContext{
+			Coinbase: addr,
+		},
+	}
+	return &msg, &scc
+}
+
+func TestStateTransition_FinalisePreviousRound(t *testing.T) {
+	currentRoundNumber := []byte("222")
+	testCoinbaseAddress := common.BigToAddress(big.NewInt(1000))
+	blockTime := big.NewInt(time.Date(2022, time.February, 28, 17, 1, 1, 0, time.UTC).Unix())
+
+	t.Run("coston chain", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("nominal case", func(t *testing.T) {
+			t.Parallel()
+
+			mockMsg, mockSCC := buildStateConnectorMock("coston chain", testCoinbaseAddress)
+
+			mockSCC.CallFunc = func(context vm.BlockContext, caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+				if caller.Address() == costonDefaultAttestors[0] {
+					rootHash := []byte("some_hash")
+					return rootHash, 0, nil
+				}
+
+				if caller.Address() != stateConnectorCoinbaseSignalAddr(params.CostonChainID, blockTime) {
+					return nil, 0, errors.New("caller address should be state connector coinbase signal addr")
```

Comment:
No need to return a real error since this is a mock. Or do we need this specific error to be returned for the test to succeed?

Reply:
Yes, this error is specified to ensure that `defer` works as expected